### PR TITLE
fix(route): use api to obtain /iwara/users

### LINKED ISSUE #12648: iwara user接口挂啦
### 路由地址

```routes
/iwara/users/:username/:type?
```

### 完整路由地址

```fullroutes
/iwara/users/kelpie/video
```

### 相关文档

https://docs.rsshub.app/en/anime.html?#iwara

### 预期是什么？

预期应当能获取到一些视频

### 实际发生了什么？

接口挂了，应该是iwara前一段时间更新过前端导致的。

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
Looks like something went wrong
Route requested: /iwara/users/kelpie/video
Error message: this route is empty, please check the original site or [create an issue](https://github.com/DIYgod/RSSHub/issues/new/choose)
Helpful Information to provide when opening issue:
Path: /iwara/users/kelpie/video
Node version: v18.16.0
Git Hash: 0f009c2
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by miles170**
## 该 PR 相关 Issue / Involved Issue

Close #12648 

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/iwara/users/kelpie/video
/iwara/users/kelpie/image
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by RimoChan** [comment]
好耶！

File: lib/v2/iwara/index.js [hunk 1]
```diff
@@ -1,43 +1,54 @@
...
-    video: 'div.node.node-video.node-teaser.node-teaser.clearfix',
-    image: 'div.node.node-image.node-teaser.node-teaser.clearfix',
+const apiUrlMap = {
+    video: `${apiRootUrl}/videos`,
+    image: `${apiRootUrl}/images`,
+};
+
+const parseThumbnail = (type, item) => {
+    if (type === 'image') {
+        return `<img src="${imageRootUrl}/image/thumbnail/${item.thumbnail.id}/${item.thumbnail.name}">`;
```
Comment (TonyRL) [inline lib/v2/iwara/index.js:20]:
```suggestion
        return `<img src="${imageRootUrl}/image/original/${item.thumbnail.id}/${item.thumbnail.name}">`;
```

File: lib/v2/iwara/index.js [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
+    }
+
+    if (item.embedUrl === null) {
+        return `<img src="${imageRootUrl}/image/thumbnail/${item.file.id}/thumbnail-${String(item.thumbnail).padStart(2, '0')}.jpg">`;
```
Comment (TonyRL) [inline lib/v2/iwara/index.js:24]:
```suggestion
        return `<img src="${imageRootUrl}/image/original/${item.file.id}/thumbnail-${String(item.thumbnail).padStart(2, '0')}.jpg">`;
```

File: lib/v2/iwara/index.js [hunk 3]
```diff
@@ -1,43 +1,54 @@
...
+    }
+
+    // regex borrowed from https://stackoverflow.com/a/3726073
+    const match = /http(?:s?):\/\/(?:www\.)?youtu(?:be\.com\/watch\?v=|\.be\/)([\w\-_]*)(&(amp;)?‌[\w?=]*)?/.exec(item.embedUrl);
+    if (match) {
+        return `<img src="${imageRootUrl}/image/embed/thumbnail/youtube/${match[1]}">`;
```
Comment (TonyRL) [inline lib/v2/iwara/index.js:30]:
```suggestion
        return `<img src="${imageRootUrl}/image/embed/original/youtube/${match[1]}">`;
```

File: lib/v2/iwara/index.js [hunk 4]
```diff
@@ -1,43 +1,54 @@
 const got = require('@/utils/got');
-const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
 
 const rootUrl = 'https://ecchi.iwara.tv';
```
Comment (TonyRL) [inline lib/v2/iwara/index.js:4]: https://ecchi.iwara.tv/users/kelpie will be redirected to https://www.iwara.tv/users/kelpie

**Pull request merged**